miio/huizuo.py

```diff
@@ -0,0 +1,172 @@
+        #########################################################################################################################
+
+        self.data = data
+
+    @property
+    def is_on(self) -> bool:
+        """Return whether the bulb is on or off."""
+        return self.data["power"]
+
+    @property
+    def brightness(self) -> int:
+        """Return current brightness."""
+        return self.data["brightness"]
+
+    @property
+    def color_temp(self) -> int:
+        """Return current color temperature."""
+        return self.data["temperature"]
+
+    def __repr__(self):
+        s = "<Huizuo on=%s brightness=%s color_temp=%s>" % (
+            self.is_on,
+            self.brightness,
+            self.color_temp,
+        )
+        return s
+
+
+class Huizuo(Device):
+    """A support for Huizuo PIS123."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_HUIZUO_PIS123,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in MODELS_SUPPORTED:
+            self.model = model
+        else:
+            self.model = MODEL_HUIZUO_PIS123
+            _LOGGER.error(
+                "Device model %s unsupported. Falling back to %s.", model, self.model
+            )
+
+    @command(
+        default_output=format_output(
+            "\n",
+            "Power: {result.is_on}\n"
+            "Brightness: {result.brightness}\n"
+            "Temperature: {result.color_temp}\n"
+            "\n",
+        )
+    )
+    def status(self) -> HuizuoStatus:
+        """Retrieve properties."""
+        properties = [
+            {"siid": 2, "piid": 1},
+            {"siid": 2, "piid": 2},
+            {"siid": 2, "piid": 3},
+        ]
+        properties_name = ["power", "brightness", "temperature"]
+
+        values = self.get_properties(properties)
+        values_info = []
+        for value in values:
+            values_info.append(value["value"])
+        return HuizuoStatus(dict(zip(properties_name, values_info)))
+
+    @command(
+        default_output=format_output("Powering on"),
+    )
+    def on(self):
+        """Power on."""
+        return self.raw_command("set_prop", [{"siid": 2, "piid": 1, "value": True}])
+
+    @command(
+        default_output=format_output("Powering off"),
+    )
+    def off(self):
+        """Power off."""
+        return self.raw_command("set_prop", [{"siid": 2, "piid": 1, "value": False}])
+
+    @command(
+        click.argument("level", type=int),
+        default_output=format_output("Setting brightness to {level}"),
+    )
+    def set_brightness(self, level):
+        """Set brightness."""
+        if level < 0 or level > 100:
+            raise HuizuoException("Invalid brightness: %s" % level)
+        return self.raw_command("set_prop", [{"siid": 2, "piid": 2, "value": level}])
```

Comment:
```suggestion
            raise HuizuoException("Invalid brightness: %s" % level)

        return self.raw_command("set_prop", [{"siid": 2, "piid": 2, "value": level}])
```
Separating the return from checks makes the code more readable.

Reply:
Changed